### PR TITLE
chore(sql): avoid indirect conversion UTF-8 -> UTF-16 -> (Long256 and GeoHash)

### PR DESCRIPTION
**Pull request by mtopolnik**
Recently added support for missing VARCHAR conversion used an indirect route by reusing the existing STRING conversion code. This adds more code for the direct route.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 49 / 58 (84.48%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WriterRowUtils.java|23|28|82.14%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineWalAppender.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/GeoHashes.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cairo/VarcharTypeDriver.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/RecordToRowCopierUtils.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cairo/GeoHashes.java [hunk 1]
```diff
@@ -204,18 +210,28 @@ public static long fromStringTruncatingNl(CharSequence hash, int start, int end,
         return widen(fromString(hash, start, start + chars), fromBits, toBits);
     }
 
-    public static long fromStringTruncatingNl(long lo, long hi, int bits) throws NumericException {
+    public static long fromAscii(Utf8Sequence hash, int start, int end) throws NumericException {
+        // no bounds/length/nullity checks
+        int size = hash.size();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/GeoHashes.java:215]: unused variable

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 144 / 150 (96.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/BindVariableServiceImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|4|7|57.14%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/std/str/Utf8s.java|44|45|97.78%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineWalAppender.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/GeoHashes.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/VarcharTypeDriver.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryPARWImpl.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WriterRowUtils.java|28|28|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/RecordToRowCopierUtils.java|14|14|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/GeoHashAdapter.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 143 / 149 (95.97%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/BindVariableServiceImpl.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|4|7|57.14%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/std/str/Utf8s.java|44|45|97.78%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineWalAppender.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/GeoHashes.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/VarcharTypeDriver.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryPARWImpl.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WriterRowUtils.java|28|28|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/RecordToRowCopierUtils.java|14|14|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/GeoHashAdapter.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**